src/common/test_helper.rs

```diff
@@ -110,128 +130,78 @@ impl FakeCrypto {
         }
         unprotected_header.rest.push((Label::Int(47), Value::Null));
         protected_header.alg = Some(coset::Algorithm::Assigned(Algorithm::Direct));
-        protected_header.key_id = key.kid.to_vec();
+        protected_header.key_id = key.key_id.clone();
         Ok(())
     }
 }
 
-#[derive(Clone)]
-pub(crate) struct FakeKey {
-    key: [u8; 5],
-    kid: [u8; 2],
-}
-
-impl ToCoseKey for FakeKey {
-    fn to_cose_key(&self) -> CoseKey {
-        CoseKeyBuilder::new_symmetric_key(self.key.to_vec())
-            .key_id(self.kid.to_vec())
-            .build()
-    }
-}
-
-impl TryFrom<Vec<u8>> for FakeKey {
-    type Error = String;
-
-    // Should be 5 bytes of key + 2 bytes of kid
-    fn try_from(mut value: Vec<u8>) -> Result<Self, Self::Error> {
-        let kid_value = value.split_off(5);
-        let key: [u8; 5] = value.try_into().map_err(|_| "Invalid input size")?;
-        let kid: [u8; 2] = kid_value.try_into().map_err(|_| "Invalid input size")?;
-        Ok(FakeKey { key, kid })
-    }
-}
-
-impl From<FakeKey> for Vec<u8> {
-    fn from(k: FakeKey) -> Self {
-        let mut key = k.key.to_vec();
-        key.append(&mut k.kid.to_vec());
-        key
-    }
-}
-
 /// Implements basic operations from the [`CoseEncryptCipher`] trait without actually using any
 /// "real" cryptography.
 /// This is purely to be used for testing and obviously offers no security at all.
 impl CoseEncryptCipher for FakeCrypto {
-    type EncryptKey = FakeKey;
-    type DecryptKey = Self::EncryptKey;
-    type Error = String;
-
     fn encrypt(
-        key: &Self::EncryptKey,
+        key: &CoseKey,
         plaintext: &[u8],
         aad: &[u8],
         protected_header: &Header,
         unprotected_header: &Header,
     ) -> Vec<u8> {
         // We put the key and the AAD before the data.
         // Again, this obviously isn't secure in any sane definition of the word.
-        let mut result: Vec<u8> = key.key.to_vec();
+        let mut result: Vec<u8> = get_symmetric_key_value(key);
         result.append(&mut aad.to_vec());
         result.append(&mut plaintext.to_vec());
         result
     }
 
     fn decrypt(
-        key: &Self::DecryptKey,
+        key: &CoseKey,
         ciphertext: &[u8],
         aad: &[u8],
         unprotected_header: &Header,
         protected_header: &ProtectedHeader,
     ) -> Result<Vec<u8>, CoseCipherError<Self::Error>> {
         // Now we just split off the AAD and key we previously put at the end of the data.
         // We return an error if it does not match.
-        if key.kid.to_vec() != protected_header.header.key_id {
+        if key.key_id.clone() != protected_header.header.key_id {
```

Comment:
Now that we have a CoseKey here, couldn't we just use a reference instead?
(then again, it's only for testing, so it probably doesn't matter)
```suggestion
        if &key.key_id != &protected_header.header.key_id {
```